cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -484,8 +484,8 @@ protected void setConfiguration(AbstractConfiguration configuration) {
         this.configuration = configuration;
     }
 
-    public String getVolumeName() {
-        return this.VOLUME_NAME;
+    protected String getVolumeName() {
+        return storage.getId() != 0 ? VOLUME_NAME + "-" + storage.getId() : VOLUME_NAME;
```

Comment:
What if the user does following:

```yaml
    storage:
      type: jbod
      volumes:
      - id: 0
        type: persistent-claim
        size: 100Gi
        deleteClaim: false
      - id: 1
        type: persistent-claim
        size: 200Gi
        deleteClaim: true
```

Then you end up with one volume named without the index and one with the index. Is that intentional?

Reply:
The code is not already there because CO is not handling jbod yet but it will raise an exception not creating the cluster if a volume has id equals to 0 (which means no id, so using `data` as volume name).

Reply:
Ah, ok. So the numbering will have to start from 1. Get it.

Reply:
I didn't want to have for the user the "zero based" array experience, but brokers and related pods with StatefulSets start already from 0 so maybe we should allow that. Now we could use `-1` as default (no id) or changing it to a `String` and then checking in the CO that it's a number. Wdyt?

Reply:
Can you not just use a (nullable) `Integer`? With a `@Minimum(0)`, because allowing negatives would be weird.

Reply:
Yeah, it's better. I already did that for some Kafka Mirror Maker consumer configuration.